File: Extensions/Tessa.Extensions.Client/UI/PnrUKSuppAgrUIExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls;

namespace Tessa.Extensions.Client.UI
{
    public sealed class PnrUKSuppAgrUIExtension : CardUIExtension
    {
        public override Task Initialized(ICardUIExtensionContext context)
        {
            Card card = context.Card;
            ICardModel cardModel = context.Model;

            // Создание карточки
            if (card.StoreMode == CardStoreMode.Insert)
            {
                // Заполнить поле Дата регистрации текущей датой
                if (cardModel.Controls.TryGet("RegistrationDate", out IControlViewModel projectDateControl))
                {
                    ((DateTimeViewModel)projectDateControl).SelectedDate = DateTime.Now;
                }
            }

            return base.Initialized(context);
        }
    }
}

File: WebClient SDK/src/PnrScripts/UI/PnrUKSuppAgrUIExtension.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { Guid } from 'tessa/platform';
import { CardStoreMode } from 'tessa/cards';
import { DateTimeViewModel } from 'tessa/ui/cards/controls';
import { PnrCardTypes } from '../Shared/PnrCardTypes';
import Moment from 'moment';

export class PnrUKSuppAgrUIExtension extends CardUIExtension {
  public initialized(context: ICardUIExtensionContext) {
    if (!Guid.equals(context.card.typeId, PnrCardTypes.PnrSupplementaryAgreementUKTypeID)) {
      return;
    }

    // Создание карточки
    if (context.card.storeMode == CardStoreMode.Insert)
    {
        const projectDateControl = context.model.controls.get('RegistrationDate') as DateTimeViewModel;
        if (!projectDateControl) return;
        // Заполнить поле Дата регистрации текущей датой
        projectDateControl.selectedDate = Moment(Date.now());
    }
  }
}